lib/upipe-av/upipe_avcodec_encode.c

```diff
@@ -509,31 +509,35 @@ static bool upipe_avcenc_encode_frame(struct upipe *upipe,
     uint8_t *buf;
     if (unlikely(!ubase_check(ubuf_block_write(ubuf, 0, &size, &buf)))) {
         ubuf_free(ubuf);
-        av_free_packet(&avpkt);
+        av_packet_unref(&avpkt);
         upipe_throw_fatal(upipe, UBASE_ERR_ALLOC);
         return false;
     }
     memcpy(buf, avpkt.data, size);
     ubuf_block_unmap(ubuf, 0);
-    av_free_packet(&avpkt);
+
+    int64_t pkt_pts = avpkt.pts, pkt_dts = avpkt.dts;
+    bool keyframe = avpkt.flags & AV_PKT_FLAG_KEY;
```

Comment:
AFAIK !! makes sure the boolean is 0 or 1 rather than something > 0 that still makes the boolean true, so it should be good if we don't write that boolean as 32 bits into a bitstream